antlr-kotlin-runtime/src/commonMain/kotlin/com/strumenta/kotlinmultiplatform/Assert.kt

```diff
@@ -3,14 +3,17 @@ package com.strumenta.kotlinmultiplatform
 import kotlin.contracts.ExperimentalContracts
 import kotlin.contracts.contract
 
+/**
+ * Throws an [AssertionError] if the [condition] is `false`.
+ */
 @OptIn(ExperimentalContracts::class)
-public inline fun assert(condition: Boolean, lazyMessage: () -> String? = { null }) {
+@Suppress("NOTHING_TO_INLINE", "KotlinRedundantDiagnosticSuppress")
+public inline fun assert(condition: Boolean, message: String? = null) {
```

Comment:
I think it was a good idea to have the message being generated lazily, as it typically done by assert functions

Reply:
@ftomassetti luckily no assertions use custom messages. We can restore the lazy message when we have assertions with custom messages, but for now it doesn't make sense imo, it's just a waste of resources especially in JS.

Reply:
@ftomassetti I'd prefer too! However our assertions here are not _real_ assertions, in the sense that they are always enabled for all platforms. Once JS gets proper support for assertions, we can use the stdlib ones and the `-ea` Kotlin compiler flags to enable/disable them.